pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/rhba/OfflineManifestGenerator.java

```diff
@@ -0,0 +1,108 @@
+package org.jboss.pnc.bacon.pig.impl.addons.rhba;
+
+import org.jboss.pnc.bacon.common.exception.FatalException;
+import org.jboss.pnc.bacon.pig.impl.addons.AddOn;
+import org.jboss.pnc.bacon.pig.impl.config.PigConfiguration;
+import org.jboss.pnc.bacon.pig.impl.pnc.ArtifactWrapper;
+import org.jboss.pnc.bacon.pig.impl.pnc.BuildInfoCollector;
+import org.jboss.pnc.bacon.pig.impl.pnc.PncBuild;
+import org.jboss.pnc.bacon.pig.impl.utils.GAV;
+import org.jboss.pnc.enums.RepositoryType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Generates the offline manifest text file, which contains all the dependencies, including third party, non-RH
+ * dependencies. First, it adds the build artifacts for all the builds. Then, it executes a query to PNC, requesting the
+ * third party dependencies for each build. Finally, the method generates a file, containing version paths and
+ * checksums, where we can get it, for all build and dependency attributes. The offline manifest file in combination
+ * with the offliner tool (https://repo1.maven.org/maven2/com/redhat/red/offliner/offliner) are used to download all the
+ * attributes to a local maven repository.
+ */
+public class OfflineManifestGenerator extends AddOn {
+
+    private static final String ADDON_NAME = "offlineManifestGenerator";
+
+    private static final Logger log = LoggerFactory.getLogger(OfflineManifestGenerator.class);
+
+    public OfflineManifestGenerator(
+            PigConfiguration pigConfiguration,
+            Map<String, PncBuild> builds,
+            String releasePath,
+            String extrasPath) {
+        super(pigConfiguration, builds, releasePath, extrasPath);
+    }
+
+    public String getName() {
+        return ADDON_NAME;
+    }
+
+    public void trigger() {
+        log.info("Will generate the offline manifest");
+        BuildInfoCollector buildInfoCollector = new BuildInfoCollector();
+        List<ArtifactWrapper> artifactsToListRaw = new ArrayList<>();
+        for (PncBuild build : builds.values()) {
+            artifactsToListRaw.addAll(build.getBuiltArtifacts());
+            // TODO: Add filter, basing on the targetRepository.repositoryType, when NCL-6079 is done
+            buildInfoCollector.addDependencies(build, "");
+            artifactsToListRaw.addAll(build.getDependencyArtifacts());
+            log.debug("Dependencies for build {}: {}", build.getId(), build.getDependencyArtifacts().size());
+        }
+        log.debug("Number of collected artifacts for the Offline manifest: {}", artifactsToListRaw.size());
+
+        List<String> exclusions = pigConfiguration.getFlow().getRepositoryGeneration().getExcludeArtifacts();
```

Comment:
is `excludeArtifacts` sth that was added for offliner or was it there before?

Reply:
It's already there in the repo generation section.